tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -331,95 +328,76 @@ def testee(a: gtx.Field[[IDim], float64]) -> gtx.Field[[IDim], np.float32]:
-    a_I_arr = np.random.randn(size, size).astype("float64")
-    a_I_float = gtx.np_as_located_field(IDim, KDim)(a_I_arr)
-    a_I_float_1 = gtx.np_as_located_field(IDim, KDim)(
-        np.append(np.insert(a_I_arr, size, 0, axis=1), [np.array([0] * (size + 1))], axis=0)
-    )
-    offset_field_arr = np.ones((size - 1, size - 1), dtype=int64)
-    offset_field_comp = np.append(
-        np.insert(offset_field_arr, size - 1, 0, axis=1), [np.array([0] * size)], axis=0
+def test_offset_field(cartesian_case):
+    ref = np.full(
+        (cartesian_case.default_sizes[IDim], cartesian_case.default_sizes[KDim]), True, dtype=bool
     )
-    offset_field = gtx.np_as_located_field(IDim, KDim)(offset_field_comp)
-    out_I_float = gtx.np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
-    out_I_float_1 = gtx.np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
 
-    @gtx.field_operator(backend=fieldview_backend)
+    @gtx.field_operator(backend=cartesian_case.backend)
     def offset_index_field_fo(
-        a: gtx.Field[[IDim, KDim], float64],
-        offset_field: gtx.Field[[IDim, KDim], int64],
-    ) -> gtx.Field[[IDim, KDim], float64]:
+        a: cases.IKField, offset_field: cases.IKField
+    ) -> gtx.Field[[IDim, KDim], bool]:
         a_i = a(as_offset(Ioff, offset_field))
         a_i_k = a_i(as_offset(Koff, offset_field))
-        return a_i_k
+        b_i = a(Ioff[1])
+        b_i_k = b_i(Koff[1])
+        return a_i_k == b_i_k
+
+    out = cases.allocate(cartesian_case, offset_index_field_fo, cases.RETURN)()
+    a = cases.allocate(cartesian_case, offset_index_field_fo, "a").extend(
+        {IDim: (0, 1), KDim: (0, 1)}
+    )()
+    offset_field = cases.allocate(cartesian_case, offset_index_field_fo, "offset_field").strategy(
+        cases.ConstInitializer(1)
+    )()
 
     offset_index_field_fo(
-        a_I_float,
+        a,
         offset_field,
-        out=out_I_float,
+        out=out,
         offset_provider={"Ioff": IDim, "Koff": KDim},
     )
 
-    @gtx.field_operator(backend=fieldview_backend)
-    def offset_index_int_fo(
-        a: gtx.Field[[IDim, KDim], float64]
-    ) -> gtx.Field[[IDim, KDim], float64]:
-        a_i = a(Ioff[1])
-        a_i_k = a_i(Koff[1])
-        return a_i_k
-
-    offset_index_int_fo(
-        a_I_float_1, out=out_I_float_1, offset_provider={"Ioff": IDim, "Koff": KDim}
-    )
-    assert np.allclose(
-        out_I_float.array()[: size - 1, : size - 1], out_I_float_1.array()[: size - 1, : size - 1]
-    )
-
+    assert np.allclose(out.array(), ref)
 
-def test_nested_tuple_return(fieldview_backend):
-    a_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    b_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    out_I_float = gtx.np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
 
-    @gtx.field_operator(backend=fieldview_backend)
+def test_nested_tuple_return(cartesian_case):
```

Comment:
This is spelling out what `verify_with_default_data` already does automatically:

```python
def test_nested_tuple_return(cartesian_case):
    @gtx.field_operator
    def pack_tuple(
        a: cases.IField, b: cases.IField
    ) -> tuple[cases.IField, tuple[cases.IField, cases.IField]]:
        return (a, (a, b))

    @gtx.field_operator
    def combine(a: cases.IField, b: cases.IField) -> cases.IField:
        packed = pack_tuple(a, b)
        return packed[0] + packed[1][0] + packed[1][1]

    cases.verify_with_default_data(cartesian_case, combine, ref=lambda a, b: a + a + b)
```